app/src/main/java/com/hedvig/app/feature/crossselling/ui/detail/CrossSellFaqActivity.kt

```diff
@@ -5,43 +5,55 @@ import android.content.Intent
 import android.os.Bundle
 import androidx.activity.compose.setContent
 import androidx.lifecycle.ViewModel
-import androidx.lifecycle.flowWithLifecycle
 import androidx.lifecycle.lifecycleScope
 import androidx.lifecycle.viewModelScope
 import arrow.core.Either
 import com.hedvig.app.BaseActivity
 import com.hedvig.app.R
 import com.hedvig.app.feature.chat.data.ChatRepository
+import com.hedvig.app.feature.crossselling.model.NavigateChat
+import com.hedvig.app.feature.crossselling.model.NavigateEmbark
 import com.hedvig.app.feature.crossselling.ui.CrossSellData
+import com.hedvig.app.feature.embark.quotecart.CreateQuoteCartUseCase
 import com.hedvig.app.feature.faq.FAQBottomSheet
 import com.hedvig.app.ui.compose.theme.HedvigTheme
 import com.hedvig.app.util.extensions.showErrorDialog
-import com.hedvig.app.util.extensions.startChat
-import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.flow.launchIn
 import kotlinx.coroutines.flow.onEach
-import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
 import org.koin.androidx.viewmodel.ext.android.viewModel
+import org.koin.core.parameter.parametersOf
 
 class CrossSellFaqActivity : BaseActivity() {
-    private val model: CrossSellFaqViewModel by viewModel()
+
+    val crossSell by lazy {
+        intent.getParcelableExtra<CrossSellData>(CROSS_SELL)
+            ?: throw IllegalArgumentException("Programmer error: CROSS_SELL not passed to ${this.javaClass.name}")
+    }
+
+    private val model: CrossSellFaqViewModel by viewModel { parametersOf(crossSell) }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
-        val crossSell = intent.getParcelableExtra<CrossSellData>(CROSS_SELL)
-            ?: throw IllegalArgumentException("Programmer error: CROSS_SELL not passed to ${this.javaClass.name}")
-
-        model.events
-            .flowWithLifecycle(lifecycle)
-            .onEach { event ->
-                when (event) {
-                    CrossSellFaqViewModel.Event.Error -> showErrorDialog(getString(R.string.component_error)) {}
-                    CrossSellFaqViewModel.Event.StartChat -> startChat()
+        model.viewState.onEach { viewState ->
```

Comment:
It's still a good idea to add that `flowWithLifecycle` in this chain to ensure that the viewState updates are read only when the App is in the foreground